### PR TITLE
simple and inefficient in-memory storage

### PR DESCRIPTION
**Pull request by maxtaco**
- useful for when SQLite isn't possible
- uses very naive data structures and lots of linear scans
- serializes all access
- passes all storage tests
- should be good enough for testing

### CODE REVIEW

File: storage/mem_sm.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+	gregor "github.com/keybase/gregor"
+	"sync"
+	"time"
+)
+
+// MemEngine is an implementation of a gregor StateMachine that just keeps
+// all incoming messages in a hash table, with one entry per user. It doesn't
+// do anything fancy w/r/t indexing Items, so just iterates over all of them
+// every time a dismissal or a state dump comes in. Used mainly for testing
+// which SQLite isn't available.
```
Comment (patrickxb) [inline storage/mem_sm.go]: which -> when

**Comment by patrickxb** [comment]
LGTM

**Comment by maxtaco** [comment]
comment fixed too

**Pull request merged**